packages/editor/src/components/post-actions/index.js

```diff
@@ -20,38 +20,62 @@ import { usePostActions } from './actions';
 
 const { Menu, kebabCase } = unlock( componentsPrivateApis );
 
-export default function PostActions( { postType, postId, onActionPerformed } ) {
-	const [ activeModalAction, setActiveModalAction ] = useState( null );
-	const { item, permissions } = useSelect(
+function useEditedEntityRecordsWithPermissions( postType, postIds ) {
+	const { items, permissions } = useSelect(
 		( select ) => {
 			const { getEditedEntityRecord, getEntityRecordPermissions } =
 				unlock( select( coreStore ) );
 			return {
-				item: getEditedEntityRecord( 'postType', postType, postId ),
-				permissions: getEntityRecordPermissions(
-					'postType',
-					postType,
-					postId
+				items: postIds.map( ( postId ) =>
+					getEditedEntityRecord( 'postType', postType, postId )
+				),
+				permissions: postIds.map( ( postId ) =>
+					getEntityRecordPermissions( 'postType', postType, postId )
 				),
 			};
 		},
-		[ postId, postType ]
+		[ postIds, postType ]
 	);
-	const itemWithPermissions = useMemo( () => {
-		return {
+
+	return useMemo( () => {
+		return items.map( ( item, index ) => ( {
 			...item,
-			permissions,
-		};
-	}, [ item, permissions ] );
+			permissions: permissions[ index ],
+		} ) );
+	}, [ items, permissions ] );
+}
+
+export default function PostActions( {
+	postType,
+	postId,
+	postIds,
+	onActionPerformed,
+} ) {
+	const [ activeModalAction, setActiveModalAction ] = useState( null );
+	const _postIds = useMemo( () => {
+		if ( postIds && postIds.length ) {
+			return postIds;
+		}
+		return postId ? [ postId ] : [];
+	}, [ postId, postIds ] );
+
+	const itemsWithPermissions = useEditedEntityRecordsWithPermissions(
+		postType,
+		_postIds
+	);
 	const allActions = usePostActions( { postType, onActionPerformed } );
 
 	const actions = useMemo( () => {
 		return allActions.filter( ( action ) => {
 			return (
-				! action.isEligible || action.isEligible( itemWithPermissions )
+				( ! action.isEligible ||
+					itemsWithPermissions.some( ( itemWithPermissions ) =>
+						action.isEligible( itemWithPermissions )
+					) ) &&
+				( itemsWithPermissions.length < 2 || action.supportsBulk )
```

Comment:
Remark: Some of this logic and the UI component itself to render the actions dropdown... is already present in the DataViews package. It seems like instead of duplicating all of that, we should be exposing a `DataActions` component from the DataViews package :)

Reply:
Yes that would make sense, as there is a lot of duplication. But it appears this was also done on purpose, given this inline comment:
https://github.com/WordPress/gutenberg/blob/trunk/packages/editor/src/components/post-actions/index.js#L88-L91
> From now on all the functions on this file are copied as from the dataviews packages,
> The editor packages should not be using the dataviews packages directly,
> and the dataviews package should not be using the editor packages directly,
> so duplicating the code here seems like the least bad option.

Should we still go ahead with this, or is there a specific reason these packages can't depend on each other?

Reply:
I don't understand why the editor shouldn't be using dataviews package. 

cc @jorgefilipecosta  

(Regardless, this should be a follow-up and not block this PR from shipping)